app/src/main/java/io/gnosis/kouban/push/PushServiceRepository.kt

```diff
@@ -81,89 +93,140 @@ class PushServiceRepository(
-        pushMessage.apply {
-            //TODO: get transaction from push message
-            val transaction = null //Transaction
-
-            val intent =
-                NavDeepLinkBuilder(context).setGraph(R.navigation.main_nav_graph).setDestination(R.id.transactionDetailsFragment).setArguments(null)
-                    .createPendingIntent()
-
-            localNotificationManager.show(
-                context,
-                context.getString(R.string.push_tex_title),
-                context.getString(R.string.push_tx_message)
+
+        lateinit var title: String
+        lateinit var text: String
+        lateinit var hash: String
+
+        when (pushMessage) {
+            is PushMessage.NewConfirmation -> {
+                hash = pushMessage.txHash
+                title = context.getString(R.string.push_tx_new_confirmation)
+                text = pushMessage.address.asEthereumAddressString()
+            }
+            is PushMessage.PendingMultisigTransaction -> {
+                hash = pushMessage.txHash
+                title = context.getString(R.string.push_tx_multisig_pending)
+                text = pushMessage.address.asEthereumAddressString()
+            }
+            is PushMessage.ExecutedMultisigTransaction -> {
+                hash = pushMessage.txHash
+                title = context.getString(R.string.push_tx_multisig_executed)
+                text = pushMessage.address.asEthereumAddressString()
+            }
+            is PushMessage.IncomingEther -> {
+                hash = pushMessage.txHash
+                title = context.getString(R.string.push_tx_incoming_ether)
+                text = pushMessage.address.asEthereumAddressString()
+            }
+            is PushMessage.IncomingToken -> {
+                hash = pushMessage.txHash
+                title = context.getString(R.string.push_tx_incoming_token)
+                text = pushMessage.address.asEthereumAddressString()
+            }
+        }
+
+        if (pushMessage is PushMessage.IncomingToken || pushMessage is PushMessage.IncomingEther) {
+
+            val intent = Intent(
+                Intent.ACTION_VIEW,
+                Uri.parse(BuildConfig.BLOCK_EXPLORER_TX.format(hash))
             )
+                .putExtra(
+                    Browser.EXTRA_APPLICATION_ID,
+                    context.packageName
+                )
+
+            localNotificationManager.show(context, 0, title, text, intent)
+
+        } else {
+
+            val intent = NavDeepLinkBuilder(context)
+                .setComponentName(MainActivity::class.java)
+                .setGraph(R.navigation.main_nav_graph)
+                .setDestination(R.id.transactionDetailsFragment)
+                .setArguments(TransactionDetailsFragmentArgs(hash).toBundle())
+                .createPendingIntent()
+
+            localNotificationManager.show(context, 0, title, text, intent, null)
         }
     }
 
-
     //TODO: get all relevant fields
```

Comment:
do we still need this reminder?

Reply:
there is a value field that we do not parse yet. Maybe we decide to show it at some point. It's a reminder that there is more stuff